### PR TITLE
import-platform: read tms file for lidar2ins transfo

### PR DESCRIPTION
**Pull request by elemoine**

### CODE REVIEW

**Comment by elemoine** [comment]
@mbredif it would be great if you could check the calculation of the affine_quat transform params. In particular I am not sure about the transform direction. I considered that the values in the tms file represent the position of the laser sensor in the ins referential. But I am not sure…

File: cli_li3ds/import_platform.py [hunk 1]
```diff
@@ -85,3 +97,29 @@ def take_action(self, parsed_args):
...
+        t1 = math.sin(heading * 0.5)
+        t2 = math.cos(roll * 0.5)
+        t3 = math.sin(roll * 0.5)
+        t4 = math.cos(pitch * 0.5)
+        t5 = math.sin(pitch * 0.5)
+        qw = t0 * t2 * t4 + t1 * t3 * t5
+        qx = t0 * t3 * t4 - t1 * t2 * t5
+        qy = t0 * t2 * t5 + t1 * t3 * t4
+        qz = t1 * t2 * t4 - t0 * t3 * t5
+
```
Comment (mbredif) [inline cli_li3ds/import_platform.py]: This TMS file seems to follow the convention Z(heading)X(roll)Y(pitch), which translates into the following quaternion (according to https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles#cite_note-nasa-rotation-2 ) :

```
h =  math.radians(heading * 0.5)
r =  math.radians(roll * 0.5)
p =  math.radians(pitch * 0.5)

ch = math.cos(h)
sh = math.sin(h)
cr = math.cos(r)
sr = math.sin(r)
cp = math.cos(p)
sp = math.sin(p)

qw = - sh * sr * sp + ch * cr * cp
qx = - sh * cr * sp + ch * sr * cp
qy = + sh * sr * cp + ch * cr * sp
qz = + ch * sr * sp + sh * cr * cp
```

but that needs testing for verification !

Comment (elemoine) [inline cli_li3ds/import_platform.py]: And Julien said that angles in the TMS file are in degrees.

Comment (elemoine) [inline cli_li3ds/import_platform.py]: Thanks.

This makes me wonder about Sbet files. Is the same convention used? I need to verify that this query is correct: https://github.com/LI3DS/api-li3ds/blob/master/api_li3ds/apis/foreignpc.py#L436-L467.

**Comment by elemoine** [comment]
Too many things/branches at the same time… So I am going to merge this, and I'll do more testing later. 

Thanks for the information provided @mbredif.

File: cli_li3ds/import_platform.py (hunk 1 above)
Comment (mbredif) [inline cli_li3ds/import_platform.py]: Hum... radian conversion is definitely missing there.

Here is the code we use to compute the rotation when using a sbet to transform a local metric frame to Lambert93 (the translation being performed by proj4)
```
    double conv_meridien = 0.72537437089 * (event.m_longitude-0.0523598775598); // sin(0.811578102)=0.72537437089
    double heading = event.m_plateformHeading-event.m_wanderAngle-conv_meridien;
    double cp = cos(event.m_pitch), sp = sin(event.m_pitch);
    double cr = cos(event.m_roll), sr = sin(event.m_roll);
    double ch = cos(heading), sh = sin(heading);
    return [      cp*sh, sr*sp*sh+cr*ch, cr*sp*sh-sr*ch,
                      cp*ch, sr*sp*ch-cr*sh, cr*sp*ch+sr*sh,
                      sp   ,-sr*cp         ,-cr*cp         ];
```

but I still have to figure out an equivalent quaternion...

Comment (elemoine) [inline cli_li3ds/import_platform.py]: I had assumed that radians were used in Sbet files.

Comment (mbredif) [inline cli_li3ds/import_platform.py]: Oops. You are right. As the code shows, no conversion is needed, they are already in radians...

**Comment by mbredif** [comment]
Sure, go ahead!

**Pull request merged**